app/workers/bgs/submit_form674_job.rb

```diff
@@ -11,23 +11,23 @@ class Invalid674Claim < StandardError; end
 
     sidekiq_options retry: false
 
-    def perform(user_uuid, saved_claim_id, vet_info)
-      user = User.find(user_uuid)
-      Rails.logger.info('BGS::SubmitForm674Job running!', { user_uuid:, saved_claim_id:, icn: user&.icn })
+    def perform(user_uuid, icn, saved_claim_id, vet_info)
+      Rails.logger.info('BGS::SubmitForm674Job running!', { saved_claim_id:, icn: })
       in_progress_form = InProgressForm.find_by(form_id: FORM_ID, user_uuid:)
       in_progress_copy = in_progress_form_copy(in_progress_form)
       claim_data = valid_claim_data(saved_claim_id, vet_info)
       normalize_names_and_addresses!(claim_data)
+      temp_user = generate_temp_user(vet_info['veteran_information'])
 
-      BGS::Form674.new(user).submit(claim_data)
-      send_confirmation_email(user)
+      BGS::Form674.new(temp_user).submit(claim_data)
+      send_confirmation_email(user_uuid, temp_user.va_profile_email, temp_user.first_name)
       in_progress_form&.destroy
-      Rails.logger.info('BGS::SubmitForm674Job succeeded!', { user_uuid:, saved_claim_id:, icn: user&.icn })
+      Rails.logger.info('BGS::SubmitForm674Job succeeded!', { user_uuid:, saved_claim_id:, icn: })
     rescue => e
-      Rails.logger.error('BGS::SubmitForm674Job failed!', { user_uuid:, saved_claim_id:, icn: user&.icn, error: e.message }) # rubocop:disable Layout/LineLength
+      Rails.logger.error('BGS::SubmitForm674Job failed!', { user_uuid:, saved_claim_id:, icn:, error: e.message })
       log_message_to_sentry(e, :error, {}, { team: 'vfs-ebenefits' })
       salvage_save_in_progress_form(FORM_ID, user_uuid, in_progress_copy)
-      DependentsApplicationFailureMailer.build(user).deliver_now if user.present?
+      DependentsApplicationFailureMailer.build(temp_user).deliver_now if temp_user.present?
```

Comment:
What should happen if `temp_user` is `nil`? Is it ever `nil` in this case?

Reply:
Good question. It looks like previously it just skipped that email from being built. I think with temp_user (not user_struct) being blank we'd do the same.

Reply:
Yeah makes sense, seems less than ideal but like you said existing behavior